src/transformers/tokenization_auto.py

```diff
@@ -233,6 +237,10 @@ def from_pretrained(cls, pretrained_model_name_or_path, *inputs, **kwargs):
                 raise ValueError("Tokenizer class {} does not exist or is not currently imported.")
             return tokenizer_class.from_pretrained(pretrained_model_name_or_path, *inputs, **kwargs)
 
+        # if model is an encoder decoder, the encoder tokenizer class is used by default
```

Comment:
This is needed so that the `EncoderDecoderModel` class can be used with `AutoTokenizer`

Reply:
great change. do we need to warn if decoder tokenizer is different?

Reply:
yeah that's a good idea!